sdk/search/Azure.Search.Documents/src/Models/SearchField.cs

```diff
@@ -1,12 +1,137 @@
+        /// </summary>
+        /// <remarks>
+        /// You can hide a field from search results if you want to use it only as a filter, for sorting, or for scoring.
+        /// This property must be true for key fields, and must be null for complex fields.
+        /// This property can also be changed on existing fields and enabling it does not cause an increase in index storage requirements.
+        /// </remarks>
+        public bool? IsHidden
+        {
+            get => IsRetrievable.HasValue ? !IsRetrievable : null;
+            set => IsRetrievable = value.HasValue ? !value : null;
+        }
+
+        [CodeGenSchemaMember("retrievable")]
+        private bool? IsRetrievable { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether the field can be referenced in a <c>$orderby</c> expression. The default is null.
+        /// A simple field can be sortable only if it is a single-valued type such as <see cref="DataType.String"/> or <see cref="DataType.Int32"/>.
+        /// </summary>
+        [CodeGenSchemaMember("sortable")]
+        public bool? IsSortable { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether the field can be retrieved in facet queries. The default is null.
+        /// This property must be null for complex fields, but can be set on simple fields within a complex field.
+        /// </summary>
+        /// <remarks>
+        /// Facets are used in presentation of search results that include hit counts by categories.
+        /// For example, in a search for digital cameras, facets might include branch, megapixels, price, etc.
+        /// </remarks>
+        [CodeGenSchemaMember("facetable")]
+        public bool? IsFacetable { get; set; }
+
+        /// <summary>
+        /// Gets or sets whether the field is the key field. The default is null.
+        /// A <see cref="SearchIndex"/> must have exactly one key field of type <see cref="DataType.String"/>.
+        /// </summary>
+        [CodeGenSchemaMember("key")]
+        public bool? IsKey { get; set; }
+
+        // TODO: Remove "overrides" for collection properties when https://github.com/Azure/autorest.csharp/issues/521 is fixed.
+
+        /// <summary>
+        /// Gets a list of names of synonym maps associated with this field. Only fields where <see cref="IsSearchable"/> is true can have associated synonym maps.
+        /// </summary>
+        [CodeGenSchemaMember("synonymMaps")]
+        public IList<string> SynonymMaps { get; internal set; }
+
+        /// <summary>
+        /// Gets a list of nested fields if this field is of type <see cref="DataType.Complex"/> or "Collection(DataType.Complex)".
+        /// </summary>
+        [CodeGenSchemaMember("fields")]
+        public IList<SearchField> Fields { get; internal set; }
+
+
+        /// <inheritdoc/>
+        public override string ToString() => $"{Name} : {Type}";
```

Comment:
Should we say `$"{{{nameof(SearchField)} {nameof(Name)}={Name}, {nameof(Type)}={Type}}}"` to make it clearer what those two values are?  We don't really have standard guidelines on how to structure format strings and I think they're starting to veer in different directions.

Reply:
I could, but the longer this gets the worse it is for debugging on smaller resolutions. This is only for debugging purposes - not a serialized form of the field.

Reply:
I guess none of the DataType names are likely to conflict with field names so it should always be obvious.

Reply:
I added the following to the XML docs, if that helps:
```csharp
        /// <remarks>
        /// This always returns "<see cref="Name"/> : <see cref="Type"/>" and is meant for debugging purposes.
        /// </remarks>
```

The goal was something short but meaningful. The object type is shown by default in the various variables windows.